micromamba/src/env.cpp

```diff
@@ -52,45 +32,7 @@ set_env_command(CLI::App* com, Configuration& config)
     init_general_options(list_subcom, config);
     init_prefix_options(list_subcom, config);
 
-    list_subcom->callback(
-        [&config]
-        {
-            const auto& ctx = config.context();
-            config.load();
-
-            EnvironmentsManager env_manager{ ctx };
-
-            if (ctx.output_params.json)
-            {
-                nlohmann::json res;
-                const auto pfxs = env_manager.list_all_known_prefixes();
-                std::vector<std::string> envs(pfxs.size());
-                std::transform(
-                    pfxs.begin(),
-                    pfxs.end(),
-                    envs.begin(),
-                    [](const mamba::fs::u8path& path) { return path.string(); }
-                );
-                res["envs"] = envs;
-                std::cout << res.dump(4) << std::endl;
-                return;
-            }
-
-            // format and print table
-            printers::Table t({ "Name", "Active", "Path" });
-            t.set_alignment(
-                { printers::alignment::left, printers::alignment::left, printers::alignment::left }
-            );
-            t.set_padding({ 2, 2, 2 });
-
-            for (auto& env : env_manager.list_all_known_prefixes())
-            {
-                bool is_active = (env == ctx.prefix_params.target_prefix);
-                t.add_row({ get_env_name(ctx, env), is_active ? "*" : "", env.string() });
-            }
-            t.print(std::cout);
-        }
-    );
+    list_subcom->callback([&config] { mamba::print_envs(config); });
```

Comment:
```suggestion
    list_subcom->callback([&config] {
        // TODO: make `Configuration::load` idempotent.
        config.load();
        mamba::print_envs(config);
    });
```

Reply:
That is typically what we want to remove, see my previous cmmments.